### PR TITLE
Add flow config, flow definitions for Emitter and Utils

### PR DESCRIPTION
**Pull request by nhunzaker**
Adds Flow. Starts preliminary flow types for Emitter and the utilities.

### CODE REVIEW

File: src/emitter.js [hunk 1]
```diff
@@ -1,22 +1,23 @@
...
+ * @flow
  * @fileoverview Emitter is an abstract class used by a few other
  * classes to communicate via events
  */
 
 import { isFunction } from './utils'
 
+type EventName = string
+type EventPayload = Array<any>
+type EventCallback = (...payload: EventPayload) => {}
```
Comment (leobauza) [inline src/emitter.js]: Wondering if we shouldn't pull all these type definitions into a `flow` folder or a `flow-typed` folder. Like [this](https://flow.org/en/docs/libdefs/creation/) suggests (I think)  or the way [vue.js does it](https://github.com/vuejs/vue/tree/dev/flow).

File: src/emitter.js [hunk 2]
```diff
@@ -126,7 +118,7 @@ class Emitter {
   /**
    * Remove all events for a given scope
    */
-  _removeScope(scope) {
+  _removeScope(scope: any) {
```
Comment (leobauza) [inline src/emitter.js]: Just pointing out `any` is unsafe. Is there a good reason for using it vs mixed (I think) https://flow.org/en/docs/types/any/

File: src/utils.js [hunk 3]
```diff
@@ -1,9 +1,17 @@
-import { castPath } from './key-path'
+/**
+ * @flow
+ */
```
Comment (leobauza) [inline src/utils.js]: Wondering if you know anything about flow working together with JSDocs? Also, can we just use `// @flow` for brevity and to not confuse the JSDocs writing? Maybe that's a silly thing, but thoughts?

File: src/emitter.js (hunk 2 above)
Comment (nhunzaker) [inline src/emitter.js]: I think think all of these just need to be mixed

File: src/utils.js (hunk 3 above)
Comment (nhunzaker) [inline src/utils.js]: I know DocumentationJS could generate JSDoc meta data from Flow types. My initial thoughts were to do something like:

```
/**
 * @fileoverview ....
 * @flow
 */
```

File: src/emitter.js (hunk 1 above)
Comment (nhunzaker) [inline src/emitter.js]: Sure. Let's give it a shot.

File: src/utils.js (hunk 3 above)
Comment (leobauza) [inline src/utils.js]: ah, gotcha. Yeah that'll work better.

**Comment by nhunzaker** [comment]
I'm going to close this out for now. Let's get jsdoc in place, then let's take a pass with flow.

**Pull request closed without merging**